JS/Lesson34/goksel/recipe-app/src/modules/recipes/NewRecipe/NewRecipeForm/index.tsx

```diff
@@ -0,0 +1,229 @@
+import { useForm } from "react-hook-form";
+import { handlestrIngredientsAndMeasures } from "./newRecipeHelper"
+import { Button, FormControl, FormLabel, InputLabel, Grid, MenuItem, Select, Typography, RadioGroup, FormControlLabel, Radio } from '@mui/material';
+import TextField from '@mui/material/TextField';
+import { DataProps } from "./newRecipeHelper";
+import { Areas, Categories } from "../../models";
+
+
+export const NewRecipe = () => {
+    const {
+        register,
+        handleSubmit,
+        formState: { errors },
+        reset
+    } = useForm<DataProps>();
+
+    const onSubmit = (data: DataProps) => {
+
+        const dateObject = new Date();
+        const formattedDate = dateObject.toISOString();
+        const ingredientsAndMeasures = handlestrIngredientsAndMeasures(data.strIngredientsAndMeasures);
+        delete data.strIngredientsAndMeasures;
+        const cleanedData = {
+            ...data,
+            strMeal: data.strMeal.trim().replace(/\s+/g, " "),
+            strDrinkAlternate: data.strDrinkAlternate ? data.strDrinkAlternate.trim().replace(/\s+/g, " ") : undefined,
+            strInstructions: data.strInstructions.trim().replace(/\s+/g, " "),
+            strTags: data.strTags.trim().replace(/\s+/g, ""),
+            strYoutube: data.strYoutube ? data.strYoutube.trim().replace(/\s+/g, " ") : undefined,
+            dateModified: formattedDate,
+            ...ingredientsAndMeasures
+        }
+        console.log(cleanedData);
```

Comment:
Don't forget to add this data to the redux store.

Reply:
Added.

Reply:
But ı am getting error on this line, ı couldn't understand why.